crates/pikelet-repl/src/lib.rs

```diff
@@ -176,86 +171,189 @@ pub fn run(opts: Opts) -> Result<(), Error> {
     Ok(())
 }
 
-fn eval_print(pikelet: &mut Driver, filemap: &FileMap) -> Result<ControlFlow, Vec<Diagnostic>> {
+#[derive(Clone)]
+enum ControlFlow {
+    Break,
+    Continue,
+}
+
+/// Commands entered in the REPL
+#[derive(Debug, Clone)]
+pub enum ReplCommand {
+    /// Evaluate a term
+    ///
+    /// ```text
+    /// <term>
+    /// ```
+    Eval(String),
+    /// Show the core representation of a term
+    ///
+    /// ```text
+    /// :core <term>
+    /// ```
+    Core(String),
+    /// Print some help about using the REPL
+    ///
+    /// ```text
+    /// :?
+    /// :h
+    /// :help
+    /// ```
+    Help,
+    /// Add a declaration to the REPL environment
+    ///
+    /// ```text
+    /// :let <name> = <term>
+    /// ```
+    Let(String, String),
+    ///  No command
+    NoOp,
+    /// Quit the REPL
+    ///
+    /// ```text
+    /// :q
+    /// :quit
+    /// ```
+    Quit,
+    /// Print the type of the term
+    ///
+    /// ```text
+    /// :t <term>
+    /// :type <term>
+    /// ```
+    TypeOf(String),
+}
+
+impl FromStr for ReplCommand {
+    type Err = Vec<Diagnostic>;
+
+    fn from_str(src: &str) -> Result<ReplCommand, Vec<Diagnostic>> {
+        use combine::char::*;
+        use combine::*;
+
+        let anys1 = || many1(any());
+        let spaces = || skip_many(space());
+        let spaces1 = || skip_many1(space());
+        let ident = || {
+            value(())
```

Comment:
`value` doesn't do anything here.

Reply:
Yeah, this was mainly for formatting purposes. Like when [using `succeed identity` in an Elm parser](https://github.com/brendanzab/elm-stlc/blob/770b3bae89315c24d3cc7d80ee4449b0fd94cbab/src/Main.elm#L274-L279).